enterprise/models.py

```diff
@@ -77,6 +77,35 @@ class Meta(object):
     history = HistoricalRecords()
     identity_provider = models.SlugField(null=True, blank=True, default=None)
     site = models.ForeignKey(Site, related_name="enterprise_customers")
+    
+    DATA_CONSENT_OPTIONAL = 'optional'
+    AT_LOGIN = 'at_login'
+    AT_ENROLLMENT = 'at_enrollment'
+    DATA_SHARING_CONSENT_CHOICES = (
+        (DATA_CONSENT_OPTIONAL, 'Optional'),
+        (AT_LOGIN, 'At Login'),
+        (AT_ENROLLMENT, 'At Enrollment'),
+    )
+    
+    enable_data_sharing_consent = models.BooleanField(
+        default=False,
+        help_text=_(
+            "This field is used to determine whether data sharing consent is enabled or "
+            "disabled of users signing in using this SSO provider. If disabled, consent "
```

Comment:
"for" not "of".

Also "this" is not an SSO provider anymore, and this setting has implications for SSO login but also for non-SSO course enrollment. So I think this should really just say "This field is used to determine when to prompt users to consent to sharing their data with this enterprise customer. Consent can be required at login, when enrolling into eligible courses, or (...explain when/how 'optional' works...)."